### PR TITLE
Improvements upstream

### PR DESCRIPTION
**Pull request by hugo-bmwcarit**

### CODE REVIEW

**Comment by pombredanne** [comment]
@hugo-bmwcarit Thanks ++ for all this! Let me review this in details tomorrow

**Comment by pombredanne** [comment]
@hugo-bmwcarit @neverpanic As a side note since I see you cleaned up some license conjunction/disjunction handling code,  see also #10 . I have a library that will integrate soon to handle properly flexible expressions parsing and a comprehensive boolean logic on these.

File: spdx/document.py [hunk 1]
```diff
@@ -17,6 +17,15 @@
 from spdx.version import Version
 
 
+def _add_parens(required, text):
+    """
+    Add parens around a license expression if `required` is True, otherwise
+    return `text` unmodified.
+    """
+    if required:
+        return '(' + str(text) + ')'
```
Comment (pombredanne) [inline spdx/document.py]: I wonder if using interpolation or format would not be nicer?
e.g. either:
- `return '(%s)' % (text)`
- `return '({text})'.format(a=text)`
and then maybe this simpler form (though the if construct is fine too):
- `return required and '({text})'.format(a=text) or text`

File: spdx/parsers/tagvaluebuilders.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def checksum_from_sha1(value):
     checksum. Returns None if doesn't match CHECKSUM_RE.
     """
     # More constrained regex at lexer level
-    CHECKSUM_RE = re.compile('SHA1:(.+)', re.UNICODE)
+    CHECKSUM_RE = re.compile('SHA1: *(.+)', re.UNICODE)
```
Comment (pombredanne) [inline spdx/parsers/tagvaluebuilders.py]: Are tabs or other whitespaces also OK there? e.g. would `CHECKSUM_RE = re.compile('SHA1:\s*(.+)', re.UNICODE)` be better or may be it does not make sense? 

And would it make sense to drop trailing or inner whitespaces entirely e.g. may be with `CHECKSUM_RE = re.compile('SHA1:\s*([\S]+)', re.UNICODE)` ?

File: spdx/utils.py [hunk 3]
```diff
@@ -114,7 +114,7 @@ def t_whitespace(self, t):
         pass
 
     def t_LICENSE(self, t):
-        r'[A-Za-z.0-9\-]+'
+        r'[A-Za-z.0-9\-+]+'
```
Comment (pombredanne) [inline spdx/utils.py]: Good catch!

File: spdx/writers/rdf.py [hunk 4]
```diff
@@ -151,9 +152,9 @@ def create_license_node(self, lic):
         Could be a single license (extracted or part of license list.) or
         a conjunction/disjunction of licenses.
         """
-        if isinstance(lic, document.LicenseConjuction):
+        if isinstance(lic, document.LicenseConjunction):
             return self.create_conjunction_node(lic)
-        elif isinstance(license, document.LicenseDisjunction):
+        elif isinstance(lic, document.LicenseDisjunction):
```
Comment (pombredanne) [inline spdx/writers/rdf.py]: Nice find!

File: stdeb.cfg [hunk 5]
```diff
@@ -0,0 +1,4 @@
+[DEFAULT]
+Depends = python-rdflib (>= 4.1.2), python-ply (>= 3.4), python-nose (>= 1.3)
+Copyright-File = LICENSE
```
Comment (pombredanne) [inline stdeb.cfg]: Ah! great... I guess we will need a machine-readable copyright file in the future then.... But doss `stddeb` require its config file to be at the root? or should we move it to a `debian` sub-directory?

Also, would it be possible to have a minimal smoke test that would validate a package creation? 

NB: this reminds me that I dislike having a hard dependency on `nose` for tests. I created #15 tp track this.

File: spdx/document.py (hunk 1 above)
Comment (neverpanic) [inline spdx/document.py]: I'd use
```python
return '({})'.format(text) if required else text
```
then.

File: spdx/parsers/tagvaluebuilders.py (hunk 2 above)
Comment (neverpanic) [inline spdx/parsers/tagvaluebuilders.py]: I agree for both.

**Comment by pombredanne** [comment]
@hugo-bmwcarit @neverpanic I merged this and added the few fixes from the review. Thank you!

**Comment by neverpanic** [comment]
@pombredanne Thanks. Sorry it fell off my radar.

**Comment by hugo-bmwcarit** [comment]
@pombredanne, thanks for merging that. I'm still testing your changes for that other project and didn't finish the review yet. For the SPDX Python tools, thanks for merging that and doing the changes by yourself.

**Comment by hugo-bmwcarit** [comment]
@neverpanic, I was supposed to take care of that, but @pombredanne did it faster than me.

**Comment by pombredanne** [comment]
@hugo-bmwcarit These were really minor fixes, do not mention it :)

**Pull request merged**